pydantic_ai_slim/pydantic_ai/messages.py

```diff
@@ -176,28 +174,30 @@ def has_content(self) -> bool:
             return bool(self.args.args_json)
 
 
+ModelResponseContentItem = Union[TextItem, ToolCall]
+
+
 @dataclass
-class ModelStructuredResponse:
-    """A structured response from a model.
+class ModelResponse:
+    """A response from a model."""
 
-    This is used either to call a tool or to return a structured response from an agent run.
-    """
+    items: list[ModelResponseContentItem]
+    """The content of the response."""
 
-    calls: list[ToolCall]
-    """The tool calls being made."""
+    role: Literal['model-response'] = 'model-response'
+    """Message type identifier, this type is available on all message as a discriminator."""
     timestamp: datetime = field(default_factory=_now_utc)
     """The timestamp of the response.
 
     If the model provides a timestamp in the response (as OpenAI does) that will be used.
     """
-    role: Literal['model-structured-response'] = 'model-structured-response'
-    """Message type identifier, this type is available on all message as a discriminator."""
 
+    @staticmethod
+    def from_text(content: str, timestamp: datetime | None = None) -> ModelResponse:
+        return ModelResponse([TextItem(content)], timestamp=timestamp or _now_utc())
```

Comment:
Happy to keep or drop this method, it gets used enough in tests/etc. that it feels worth keeping but I'm open to disagreement.

Reply:
decision: keep